### PR TITLE
Experiment: using capabilities for the semgrep codebase

### PR DESCRIPTION
**Pull request by aryx**
See the talk I'll make to the PA/SR talk for the motivations and explanations of this work.

### CODE REVIEW

**Review by mjambon — CHANGES_REQUESTED** [review]
I approve the new modules that end with an underscore (`List_` etc.) but not the code related to capabilities in its current design. I'll give details on this on Slack or in person.

**Comment by emjin** [comment]
Do you have a sense of what the slowdowns are coming from?

**Comment by aryx** [comment]
nope, dunno why we regress, which is why I'm merging parts of this PR piece by piece.
I'm down to +770-650 for the PR so this becomes easier to debug.

File: src/main/dune [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+    TCB
     semgrep_core_cli
     osemgrep_cli
     lwt_platform.unix
  )
  (preprocess (pps ppx_profiling))
  ; for ocamldebug
- (modes native byte)
- (flags (:include flags.sexp))
+ ;XXX(modes native byte)
```
Comment (aryx) [inline src/main/dune]: @emjin mode natives strike again!
I commented it while experimenting and forgot about it.

**Pull request merged**